src/main/java/com/google/sps/servlets/SyncServlet.java

```diff
@@ -0,0 +1,58 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.sps.data.*;
+
+/** Servlet that takes in the user's inputted Room ID and directs them
+    to the associated private room once ensurred that it's an actual room*/
+@WebServlet("/sync-room")
+public final class SyncServlet extends HttpServlet {
+    private static final String UPDATE_STATE_PARAMETER = "userState";
+    private static final String ROOM_ID_PARAMETER = "roomId";
+    private static final String VIDEO_TIMESTAMP_PARAMETER = "timeStamp";
+    @Override
+    public void doGet(HttpServletRequest req, HttpServletResponse res) throws IOException {
+        long roomId = Long.parseLong(req.getParameter(ROOM_ID_PARAMETER));
+        Room updatedRoom = Room.fromRoomId(roomId);
+        Video currentVideo = updatedRoom.getCurrentVideo();
+        String responseBody = videoObjectToJsonString(currentVideo);
+        res.setContentType(ServletUtil.JSON_CONTENT_TYPE);
+        res.getWriter().println(responseBody);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest req, HttpServletResponse res) throws IOException {
+        long roomId = Long.parseLong(req.getParameter(ROOM_ID_PARAMETER));
+        Video.VideoState newState = Video.VideoState.fromInt(Integer.parseInt(req.getParameter(UPDATE_STATE_PARAMETER)));
+        long currentVideoTimestamp = Long.parseLong(req.getParameter(VIDEO_TIMESTAMP_PARAMETER));
+
+        Room syncRoom = Room.fromRoomId(roomId);
+
+        if(newState == Video.VideoState.ENDED){
+            syncRoom.changeCurrentVideo();
+        } else {
+            syncRoom.updateCurrentVideoState(newState, currentVideoTimestamp);
+        }
+        syncRoom.toDatastore();
+        res.setStatus(200);
```

Comment:
define status constant and use it here.

Reply:
This wasn't my code, sorry for adding it in the PR